storage/src/vespa/storage/bucketdb/storagebucketdbinitializer.cpp

```diff
@@ -473,15 +509,15 @@ StorageBucketDBInitializer::sendReadBucketInfo(spi::PartitionId disk)
     NextBucketOnDiskFinder finder(disk, state._databaseIterator, count);
     LOG(spam, "Iterating bucket db further. Starting at iterator %s",
         state._databaseIterator.toString().c_str());
-    _system._bucketDatabase.all(finder,
+    _system.getBucketDatabase(bucketSpace).all(finder,
                                 "StorageBucketDBInitializer::readBucketInfo",
                                 state._databaseIterator.stripUnused().toKey());
     if (finder._alreadySet > 0) {
         _metrics._infoSetByLoad.inc(finder._alreadySet);
         _state._infoSetByLoad += finder._alreadySet;
     }
     for (uint32_t i=0; i<finder._next.size(); ++i) {
-        document::Bucket bucket(BucketSpace::placeHolder(), finder._next[i]);
+        document::Bucket bucket(bucketSpace, finder._next[i]);
         ReadBucketInfo::SP cmd(new ReadBucketInfo(bucket));
```

Comment:
Consider shoveling this into a `make_shared` as well. We'll be making a bunch of these messages, so the less allocs, the better 🦄